internal/controller/controllers/cluster_controller.go

```diff
@@ -89,9 +95,36 @@ func (r *ClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return result, err
 	}
 
+	if r.isReadyForInstallation(cluster, c) {
+		var ic *common.Cluster
+		ic, err = r.Installer.InstallClusterInternal(ctx, installer.InstallClusterParams{
+			ClusterID: *c.ID,
+		})
+		if err != nil {
+			return r.updateState(ctx, cluster, c, err)
+		}
+		return r.updateState(ctx, cluster, ic, nil)
+	}
+
 	return r.updateState(ctx, cluster, c, nil)
 }
 
+func (r *ClusterReconciler) isReadyForInstallation(cluster *adiiov1alpha1.Cluster, c *common.Cluster) bool {
+	if ready, _ := r.ClusterApi.IsReadyForInstallation(c); !ready {
+		return false
+	}
+
+	readyHosts := 0
+	for _, h := range c.Hosts {
+		if r.HostApi.IsInstallable(h) {
+			readyHosts += 1
+		}
+	}
+
+	expectedHosts := cluster.Spec.ProvisionRequirements.ControlPlaneAgents + cluster.Spec.ProvisionRequirements.WorkerAgents
```

Comment:
how can you determine by this check if the number of ready masters and workers overlaps to the internal cluster check if you check the total sum?
or is it doesn't matter since you verified it with the cluster API?